src/schema.js

```diff
@@ -22,8 +22,10 @@ const addDirectives = (schema) => {
       if (operation === 'Mutation') {
         const next = field.resolve
         field.resolve = (object, params, context, info) => {
-          // verify request with a generated scope, e.g. Mutation:DeleteControlAction, Mutation.CreatePerson, ...
-          verifyRequest(context, `${operation}:${fieldName}`)
+          // verify request with a generated scope, e.g. Mutation:ControlAction:Delete, Mutation:Person:Create, ...
+          const [, action, type] = fieldName.match(/^([A-Z][a-z]+)([A-Z]\w+)/)
```

Comment:
From what I understand here, you're taking a mutation name, e.g. `AddActionInterfaceLocation` or `CreatePerson` and splitting it on first word (action) and rest of the word (type), then re-ordering it so that we get the hierarchical operation:type:action below for the verification.
I've tested this with a few mutations and it looks fine. The only thing that I'd suggest is a specific comment that has an example of how the fieldName is re-ordered, and perhaps some tests here to check that fieldName -> reorder -> verify works with some different scopes.

Reply:
Yes, that is correct. I will add some unit tests for that feature.